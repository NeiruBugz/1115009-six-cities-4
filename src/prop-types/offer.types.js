import PropTypes from 'prop-types';

export const offerShape = PropTypes.shape({
  mark: PropTypes.string,
  image: PropTypes.string,
  offerImages: PropTypes.arrayOf(PropTypes.string),
  price: PropTypes.number,
  priceText: PropTypes.string,
  rating: PropTypes.number,
  ratingValue: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  bedroomsCount: PropTypes.number,
  adultsCount: PropTypes.number,
  features: PropTypes.arrayOf(PropTypes.string),
  hostName: PropTypes.string,
  hostAvatar: PropTypes.string,
  super: PropTypes.bool,
  offerDescription: PropTypes.arrayOf(PropTypes.string),
  coordinates: PropTypes.arrayOf(PropTypes.number),
});

export const offerList = PropTypes.arrayOf(offerShape);
