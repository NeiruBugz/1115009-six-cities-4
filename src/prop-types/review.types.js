import PropTypes from 'prop-types';

export const reviewShape = PropTypes.shape({
  userAvatar: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
});

export const reviewList = PropTypes.arrayOf(reviewShape);
