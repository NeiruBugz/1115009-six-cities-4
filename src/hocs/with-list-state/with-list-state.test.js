import React from 'react';
import {string} from 'prop-types';
import {create} from 'react-test-renderer';
import withListState from './with-list-state';

const MockedComponent = ({className}) => (
  <div className={className}>test</div>
);

MockedComponent.propTypes = {
  className: string.isRequired,
};

const MockedHOC = withListState(MockedComponent);

describe(`withListState HOC unit tests`, () => {
  it(`should render wrapped component correctly`, () => {
    const hoc = create(
        <MockedHOC className={`test`}/>
    ).toJSON();

    expect(hoc).toMatchSnapshot();
  });
});
