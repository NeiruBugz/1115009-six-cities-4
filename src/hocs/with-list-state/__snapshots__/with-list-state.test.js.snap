// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withListState HOC unit tests should render wrapped component correctly 1`] = `
<div
  className="test"
>
  test
</div>
`;
